File: battle_real_time/lib/battle_real_time/contracts/publishers/player_actions/get_team_details_contract.ex
defmodule BattleRealTime.Contracts.Publishers.PlayerActions.GetTeamDetailsContract do
  @moduledoc """
  Validates the payload for the get_team_details action.
  """
  use BattleRealTime.Contracts.Contract

  @primary_key false
  embedded_schema do
    field(:player_id, :string)
    field(:team_id, :integer)
    field(:timestamp, :string)
  end

  def changeset(struct, params) do
    struct
    |> cast(params, [:player_id, :team_id, :timestamp])
    |> validate_required([:player_id, :team_id])
  end
end

File: battle_engine/app/contracts/consumers/get_team_details_contract.rb
# frozen_string_literal: true

module Contracts
  module Consumers
    class GetTeamDetailsContract < Dry::Validation::Contract
      params do
        required(:player_id).filled(:integer)
        required(:team_id).filled(:integer)
        optional(:timestamp).maybe(:string)
      end
    end
  end
end
